pgrx/src/spi.rs

```diff
@@ -952,22 +951,26 @@ impl<'a> Query for PreparedStatement<'a> {
         (&self).execute(client, limit, arguments)
     }
 
-    fn open_cursor<'c: 'cc, 'cc>(
+    fn open_cursor(
         self,
-        client: &'cc SpiClient<'c>,
+        client: &'conn SpiClient<'conn>,
         args: Self::Arguments,
-    ) -> SpiCursor<'c> {
+    ) -> SpiCursor<'conn> {
         (&self).open_cursor(client, args)
     }
 }
 
-impl<'a> SpiClient<'a> {
+impl<'conn> SpiClient<'conn> {
     /// Prepares a statement that is valid for the lifetime of the client
     ///
     /// # Panics
     ///
     /// This function will panic if the supplied `query` string contained a NULL byte
-    pub fn prepare(&self, query: &str, args: Option<Vec<PgOid>>) -> Result<PreparedStatement> {
+    pub fn prepare(
+        &self,
+        query: &str,
+        args: Option<Vec<PgOid>>,
+    ) -> Result<PreparedStatement<'conn>> {
```

Comment:
If `SpiClient<'conn>`, then that implies SpiClient is bound by the `'conn` lifetime... so this should correctly relate the `&self` and `PreparedStatement<'conn>` lifetimes. But then there's `&str`, which implies another lifetime.

I think normal function variance should take care of how it only needs to outlive `'conn`, but it will also be easy to add another lifetime parameter later if it doesn't.

Reply:
`.prepare()` isn't also borrowing from `&self`, so I think this is okay.  IOW, it doesn't matter how long `&self` lives after we get the resulting `PreparedStatement<'conn>`.  What matter is only that the prepared statement can't somehow escape the 'conn.  KAAAAHN!!!!